google/cloud/pubsub/internal/tracing_batch_callback_test.cc

```diff
@@ -104,6 +105,40 @@ TEST(TracingBatchCallback, StartAndEndMessage) {
                             SpanHasEvents(EventNamed("gl-cpp.ack_end")))));
 }
 
+TEST(TracingBatchCallback, VerifySpanIsSetInUserCallback) {
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<pubsub_testing::MockBatchCallback>();
+  EXPECT_CALL(*mock, callback).Times(1);
+  // Since there is no default constructor, we need to create a fake before we
+  // get the actual value.
+  MessageCallback::MessageAndHandler actual_message_and_handler{
+      pubsub::MessageBuilder().Build(),
+      std::make_unique<pubsub_testing::MockExactlyOnceAckHandlerImpl>(),
+      "invalid-ack-id", Span{}};
+  EXPECT_CALL(*mock, user_callback)
+      .WillOnce([&](MessageCallback::MessageAndHandler arg0) {
+        actual_message_and_handler = std::move(arg0);
+      });
+  auto batch_callback = MakeTestBatchCallback(std::move(mock));
+  MessageCallback::MessageAndHandler message_and_handler{
+      pubsub::MessageBuilder().Build(),
+      std::make_unique<pubsub_testing::MockExactlyOnceAckHandlerImpl>(),
+      "ack-id-0", Span{}};
+
+  batch_callback->callback(MakeResponse(1));
+  batch_callback->user_callback(std::move(message_and_handler));
+  batch_callback->AckEnd("ack-id-0");
+
+  auto spans = span_catcher->GetSpans();
+  EXPECT_THAT(
+      spans, Contains(AllOf(SpanHasInstrumentationScope(), SpanKindIsConsumer(),
+                            SpanNamed("test-sub subscribe"),
+                            SpanHasEvents(EventNamed("gl-cpp.ack_end")))));
+  EXPECT_EQ(actual_message_and_handler.ack_id, "ack-id-0");
+  EXPECT_THAT(actual_message_and_handler.subscribe_span.span,
+              testing::NotNull());
```

Comment:
nit: we do not write `testing::*` in our expectations so that they read more like english.

Add a `using ::testing::NotNull;` above.

Reply:
Done